evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py

```diff
@@ -385,50 +454,149 @@ def _decompose_target(X, y, fh):
-
-        if not self.forecast_summary or len(y) != len(
-            self.forecast_summary.predicted_mean,
-        ):
-            self._project_trend_and_seasonality(y)
-
-        prediction_interval_result = {}
-        for i, alpha in enumerate(alphas):
-            result = self.forecast_summary.summary_frame(alpha=alpha)
-            overlapping_ind = [ind for ind in y.index if ind in result.index]
-            intervals = pd.DataFrame(
-                {
-                    "lower": result["mean_ci_lower"] - result["mean"],
-                    "upper": result["mean_ci_upper"] - result["mean"],
-                },
-            )
-            if len(overlapping_ind) > 0:  # y.index is datetime
-                intervals = intervals.loc[overlapping_ind]
-            else:  # y.index is not datetime (e.g. int)
-                intervals = intervals[-len(y) :]
-                intervals.index = y.index
-            prediction_interval_result[f"{coverage[i]}_lower"] = intervals["lower"]
-            prediction_interval_result[f"{coverage[i]}_upper"] = intervals["upper"]
-
-        return prediction_interval_result
+        series_results = {}
+        for id in y.columns:
+            y_series = y[id]
+
+            if coverage is None:
+                coverage = [0.95]
+
+            self._check_oos_past(y_series)
+            alphas = [1 - val for val in coverage]
+
+            if not self.forecast_summary or len(y_series) != len(
+                self.forecast_summary.predicted_mean,
+            ):
+                self._project_trend_and_seasonality(y_series)
+
+            prediction_interval_result = {}
+            for i, alpha in enumerate(alphas):
+                result = self.forecast_summary.summary_frame(alpha=alpha)
+                overlapping_ind = [ind for ind in y_series.index if ind in result.index]
+                intervals = pd.DataFrame(
+                    {
+                        "lower": result["mean_ci_lower"] - result["mean"],
+                        "upper": result["mean_ci_upper"] - result["mean"],
+                    },
+                )
+                if len(overlapping_ind) > 0:  # y.index is datetime
+                    intervals = intervals.loc[overlapping_ind]
+                else:  # y.index is not datetime (e.g. int)
+                    intervals = intervals[-len(y_series) :]
+                    intervals.index = y_series.index
+                prediction_interval_result[f"{coverage[i]}_lower"] = intervals["lower"]
+                prediction_interval_result[f"{coverage[i]}_upper"] = intervals["upper"]
+            series_results[id] = prediction_interval_result
+
+            # only return the dictionary if single series
+            if len(y.columns) <= 1:
+                return prediction_interval_result
+        return series_results
+
+    # Overload the plot_decomposition fucntion to be able to plot multiple decompositions for multiseries
+    def plot_decomposition(
```

Comment:
I don't think we need to overload the parent function here - most of this code is identical to `Decomposer.plot_decomposition`. We can update the parent class to handle this case instead, there's no need to copy code in order to keep all the changes to this one file!

I think this also ties in with changing the use of class attributes. The goal here is to update the STLDecomposer to support multiseries, because that's the decomposer implementation that we have that works, but ideally it won't be the only one forever. I think we shouldn't be afraid of updating the parent Decomposer code as well, so that if we add a new decomposer (or get the Polynomial Decomposer working, lol) it will be easier to have them also support multiseries. No need to shoehorn ourselves in here.